index.js

```diff
@@ -21,4 +22,54 @@ function hideMenu() {
 
 document.getElementById('hamburger-id').addEventListener('click', showMenu);
 document.getElementById('cancel-id').addEventListener('click', hideMenu);
-document.querySelector('.menu-list').addEventListener('click', hideMenu);
\ No newline at end of file
+document.querySelector('.menu-list').addEventListener('click', hideMenu);
+
+// Email validatetion
+const myform = document.getElementById('myform');
+const email = document.getElementById('email');
+const error = document.querySelector('.error-msg');
+const submitbtn = document.getElementById('submitbtn');
+// the function below check if the email address is all lowercase or not //
+function emailCheckLowerCase() {
+  let result = false;
+  if (email.value !== email.value.toLowerCase()) {
+    error.classList.remove('hide');
+    submitbtn.disable = true;
+  } else {
+    error.classList.add('hide');
+    result = true;
+    submitbtn.disable = false;
+  }
+  return result;
+}
+
+// local storage
+const txtname = document.getElementById('txt-name');
+const comment = document.getElementById('txt-text');
+function storeToLocalStorage() {
+  const data = {
+    email: email.value,
+    name: txtname.value,
+    comment: comment.value,
+  };
+  localStorage.setItem('formdata', JSON.stringify(data));
+}
+submitbtn.onclick = function () {
+  if (emailCheckLowerCase() === true) {
+    myform.submit();
+    storeToLocalStorage();
```

Comment:
- [ ] Your functionality does not work as expected where the user data gets saved to local storage as the user is typing because you're wrapping your code to store user data in a submit function and it won't work until the user submits the form
- [ ] I kindly suggest that you listen for changes in your inputs with an input event listener like below:

### EXAMPLE
```
const nameField = document.getElementById('userName');
const emailField = document.getElementById('userEmail');
const textField = document.getElementById('userText');

let formInfo = {
  name: '',
  email: '',
  text: '',
};

if (localStorage.formData) {
  formInfo = JSON.parse(localStorage.formData);
  nameField.value = formInfo.name;
  emailField.value = formInfo.email;
  textField.value = formInfo.text;
}

nameField.addEventListener('input', () => {
  formInfo.name = nameField.value;
  localStorage.setItem('formData', JSON.stringify(formInfo));
});

emailField.addEventListener('input', () => {
  formInfo.email = emailField.value;
  localStorage.setItem('formData', JSON.stringify(formInfo));
});

textField.addEventListener('input', () => {
  formInfo.text = textField.value;
  localStorage.setItem('formData', JSON.stringify(formInfo));
});
```